# HG changeset patch
# User Rik <rik@octave.org>
# Date 1491426678 25200
#      Wed Apr 05 14:11:18 2017 -0700
# Node ID 5ea6c6d0c2dbc0914604b03a0fae43c1fe84b5cf
# Parent  0f18524973eb16b1dbf28350819284686628b891
Use octave_base_value::print_name_tag for old style classes (bug #50640).

* ov-class.cc (octave_class::print_name_tag): Remove code for overloading print
and call octave_base_value::print_name_tag() instead.

* scripts/@ftp/disp.m, scripts/audio/@audioplayer/disp.m,
scripts/audio/@audiorecorder/disp.m: Remove trailing newline from disp()
routines for old-style classes.

diff --git a/libinterp/octave-value/ov-class.cc b/libinterp/octave-value/ov-class.cc
--- a/libinterp/octave-value/ov-class.cc
+++ b/libinterp/octave-value/ov-class.cc
@@ -992,25 +992,17 @@ octave_class::print_raw (std::ostream& o
   indent (os);
   os << "  <class " << class_name () << ">";
   newline (os);
 }
 
 bool
 octave_class::print_name_tag (std::ostream& os, const std::string& name) const
 {
-  bool retval = false;
-
-  indent (os);
-  os << name << " =";
-  newline (os);
-  if (! Vcompact_format)
-    newline (os);
-
-  return retval;
+  return octave_base_value::print_name_tag (os, name);
 }
 
 void
 octave_class::print_with_name (std::ostream& os, const std::string& name,
                                bool print_padding)
 {
   octave_base_value::print_with_name (os, name, print_padding);
 }
diff --git a/scripts/@ftp/disp.m b/scripts/@ftp/disp.m
--- a/scripts/@ftp/disp.m
+++ b/scripts/@ftp/disp.m
@@ -21,11 +21,11 @@ function disp (obj)
   if (nargin != 1)
     print_usage ();
   endif
 
   printf ("  FTP object\n");
   printf ("    host: %s\n", obj.host);
   printf ("    user: %s\n", obj.username);
   printf ("     dir: %s\n", __ftp_pwd__ (obj.curlhandle));
-  printf ("    mode: %s\n\n", __ftp_mode__ (obj.curlhandle));
+  printf ("    mode: %s\n", __ftp_mode__ (obj.curlhandle));
 
 endfunction
diff --git a/scripts/audio/@audioplayer/disp.m b/scripts/audio/@audioplayer/disp.m
--- a/scripts/audio/@audioplayer/disp.m
+++ b/scripts/audio/@audioplayer/disp.m
@@ -26,12 +26,10 @@ function disp (player)
   if (nargin != 1)
     print_usage ();
   endif
 
   printf ("audioplayer object with properties:\n\n");
   for [val, prop] = __get_properties__ (player)
     printf ("  %s = ", prop), disp (val);
   endfor
-  ## FIXME: display in pr-output.cc should add "\n", not this function. 
-  printf ("\n");
 
 endfunction
diff --git a/scripts/audio/@audiorecorder/disp.m b/scripts/audio/@audiorecorder/disp.m
--- a/scripts/audio/@audiorecorder/disp.m
+++ b/scripts/audio/@audiorecorder/disp.m
@@ -26,12 +26,10 @@ function disp (recorder)
   if (nargin != 1)
     print_usage ();
   endif
 
   printf ("audiorecorder object with properties:\n\n");
   for [val, prop] = __get_properties__ (recorder)
     printf ("  %s = ", prop), disp (val);
   endfor
-  ## FIXME: display in pr-output.cc should add "\n", not this function. 
-  printf ("\n");
 
 endfunction
